### PR TITLE
Story integration tests

### PR DESCRIPTION
**Pull request by andrewscfc**
No issue

**Overall change:**
A set of integration tests for Story pages adapted from https://github.com/bbc/simorgh/tree/latest/src/integration/pages/articlePage

Take a look at this readme to explain what integration tests are and how they are intended to be implemented: https://github.com/bbc/simorgh/blob/latest/src/integration/README.md

Targets this article on the mundo service: http://localhost:7080/mundo/noticias-internacional-51266689

**Code changes:**
Added a `src/integration/pages/storyPage` directory and implemeneted mundo tests

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/integration/pages/storyPage/user.amp.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default () => {
+  it('I can see an amp image', () => {
```
Comment (sareh) [inline src/integration/pages/storyPage/user.amp.js:2]: This seems to be repeated across the various `/integration/pages/**/user.amp.js` files - can we have it in a common file and import it where it's needed?

File: src/integration/pages/storyPage/user.canonical.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default ({ imageAltText, imageCaptionText }) => {
```
Comment (sareh) [inline src/integration/pages/storyPage/user.canonical.js:1]: Same with this function here - can we have it defined in `/integration/common/user.canonical.js` and re-use it for each page type?

File: src/integration/pages/storyPage/user.amp.js (hunk 1 above)
Comment (karinathomasbbc) [reply src/integration/pages/storyPage/user.amp.js:2]: @jroebu14 commented on a PR of mine for MAP integration tests - https://github.com/bbc/simorgh/pull/5880#pullrequestreview-378904399

TLDR;
I prefer to keep things explicit rather than make abstractions that don't abstract all that much. This goes against the DRY principle but I think we should "prefer duplication over the wrong abstraction" as discussed here https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction

Comment (sareh) [reply src/integration/pages/storyPage/user.amp.js:2]: @karinathomasbbc @jroebu14 Could you then explain why the analytics tests in these files are okay to be pulled out into a `common` directory, but not the user tests? 
I think I'm missing something here. 
`/integration/common/analytics.amp.js`
`/integration/common/analytics.canonical.js`

Comment (karinathomasbbc) [reply src/integration/pages/storyPage/user.amp.js:2]: I guess if it will work across all page types, then it would be OK to put it into the common tests - just not sure if this would work for MAPs?

Comment (sareh) [reply src/integration/pages/storyPage/user.amp.js:2]: @karinathomasbbc Ah, okay, I see. So the issue is that the image isn't common to *every single* pagetype. Happy for things to stay this way in that case.

**Review by sareh — APPROVED** [review]
👍 Looks good to me. Thanks for adding these tests, Andrew.

File: src/integration/pages/storyPage/mundo.test.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  describe('When I am analysing user/performance metrics', () => {
+    runCommonCanonicalAnalyticsTests();
+  });
+
+  describe('When the application starts', () => {
+    runCoreCanonicalTests({ service: 'mundo' });
+  });
+});
+
+describe('Given I am on a Mundo article AMP/Canonical page', () => {
```
Comment (Simon2828) [inline src/integration/pages/storyPage/mundo.test.js]: story page rather than article

File: src/integration/pages/storyPage/mundo.test.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    });
+
+    describe('When optimising the application performance', () => {
+      runPerformaceTests();
```
Comment (Simon2828) [inline src/integration/pages/storyPage/mundo.test.js]: typo `runPerformanceTests()`

File: src/integration/pages/storyPage/mundo.test.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  describe('When the application starts', () => {
+    runCoreAmpTests();
+  });
+
+  describe('When I am analysing user/performance metrics', () => {
+    runCommonAmpAnalyticsTests();
+  });
+});
+
+describe('Given I am on a Mundo stor Canonical page', () => {
```
Comment (12) [inline src/integration/pages/storyPage/mundo.test.js]:
```suggestion
describe('Given I am on a Mundo story Canonical page', () => {
```

File: src/integration/pages/storyPage/mundo.test.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import {
+  runFooterTests,
+  runHeaderTests,
+  runCommonSeoTests,
+  runCommonCanonicalAnalyticsTests,
+  runCommonAmpAnalyticsTests,
+  runCommonA11yTests,
+  runCoreAmpTests,
+  runCoreCanonicalTests,
+  runPerformaceTests,
```
Comment (HarveyPeachey) [inline src/integration/pages/storyPage/mundo.test.js]:
```suggestion
  runPerformanceTests,
```

**Review by jamesdonoh — APPROVED** [review]
Thanks for getting us going with these Andrew. All looks good to me except for the typos identified above.

**Review by simonsinclair — DISMISSED** [review]

**Review by simonsinclair — COMMENTED** [review]
Following a merge, this is failing with the following two errors.

The note in the error eludes to the cause being text broken up by multiple elements. This is the markup as it stands:
```html
<p class="StyledParagraph-sc-1gggobt-2 jQPFnr">
  <span lang="en-GB">©<!-- --> </span>2020 BBC. La BBC no se hace responsable del contenido de sitios externos.<!-- --> 
  <a href="https://www.bbc.co.uk/help/web/links/" class="StyledLink-nhmg8j-0 jHQlUp">
    <span class="StyleLinkText-nhmg8j-1 fPnERB">Lee sobre nuestra postura acerca de enlaces externos.</span>
  </a>
</p>
```

And this is the text the test is searching for:

```
 © 2020 BBC. La BBC no se hace responsable del contenido de sitios externos. Lee sobre nuestra postura acerca de enlaces externos.
```

### Errors

```
● Given I am on a Mundo story AMP/Canonical page › When I am using the website › And using Canonical › I can see the footer copyright and external linking text

    TestingLibraryElementError: Unable to find an element with the text: (content, node) => {
        const hasText = ({ textContent }) => textContent === text;
        const nodeHasText = hasText(node);
        const childrenDontHaveText = Array.from(node.children).every(
          (child) => !hasText(child),
        );

        return nodeHasText && childrenDontHaveText;
      }. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.
```

```
● Given I am on a Mundo story AMP/Canonical page › When I am using the website › And using AMP › I can see the footer copyright and external linking text

    TestingLibraryElementError: Unable to find an element with the text: (content, node) => {
        const hasText = ({ textContent }) => textContent === text;
        const nodeHasText = hasText(node);
        const childrenDontHaveText = Array.from(node.children).every(
          (child) => !hasText(child),
        );

        return nodeHasText && childrenDontHaveText;
      }. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.
```

**Comment by simonsinclair** [comment]
@jroebu14 spotted a change in API on one of the functions we were calling. I've pushed a change that should fix the failures. 🤞
